packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-groups.tsx

```diff
@@ -0,0 +1,229 @@
+import { Component, h, JSX, Element, State, readTask } from '@stencil/core';
+import { afterNextRenderTimeout } from '@utils/dom/after-next-render';
+import { trackComponent } from '@utils/tracking/usage';
+import { GuxAvatarAccent } from './../gux-avatar/gux-avatar.types';
+
+/**
+ * @slot - Some gux-avatars
+ */
+
+@Component({
+  styleUrl: 'gux-avatar-group.scss',
+  tag: 'gux-avatar-group-beta',
+  shadow: true
+})
+export class GuxAvatarGroup {
+  @Element()
+  root: HTMLElement;
+
+  private resizeObserver?: ResizeObserver;
+  private overflowAvatars: HTMLGuxAvatarFocusableBetaElement[];
+  private listItemClickListeners: Map<HTMLElement, (event: Event) => void> =
+    new Map();
+
+  @State()
+  avatarList: HTMLGuxAvatarFocusableBetaElement[];
+
+  @State()
+  private overflowCount: number = 0;
+
+  get countAvatars() {
+    return this.avatarList.length ?? 0;
+  }
+
+  private checkGroupContainerWidthForLayout() {
+    readTask(() => {
+      const container = this.root.shadowRoot.querySelector('.gux-avatar-group');
+      const containerWidth = container.clientWidth;
```

Comment:
You might be able to do this in a programmatic way as follows. Here is what I'm thinking
1. remove the hard coded `display:none` breakpoints in the scss file
2.      
Here where you check the containerWidth, should be able to compare it with the scrollWidth to programmatically show and hide avatars
```
const containerWidth = container.clientWidth;
const scrollWidth = container.scrollWidth;
// if scrollWidth > containerWidth remove the last child avatar and recalculate until scrollWidth < containerWidth
      ```
Using the above logic you could add a css class to the avatars you want to hide, a class that has the property `display: none` (instead of defining the breakpoints in the scss file) and add to the tally of `overflowAvatars`

Reply:
I agree with @katie-bobbe-genesys  on this as I think the hiding and showing of avatars plus updating of the overflow counter could be done in one file. You might also run into issues aswell with the container queries for example if there can be an unlimited number of `gux-avatars` within a group then the container query will have to be more dynamic to handle that scenario but if theres is a fixed amount it would be fine. Katies idea seems good to me by conditionally adding a class based on the breakpoint widths which you have in the `getMaxAvatarsToDisplay` function already.

Reply:
I also agree with @katie-bobbe-genesys here. Great thinking.